certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/message/service/DeleteAnswerService.java

```diff
@@ -0,0 +1,57 @@
+package se.inera.intyg.certificateservice.application.message.service;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.certificateservice.application.common.ActionEvaluationFactory;
+import se.inera.intyg.certificateservice.application.message.dto.DeleteAnswerRequest;
+import se.inera.intyg.certificateservice.application.message.dto.DeleteAnswerResponse;
+import se.inera.intyg.certificateservice.application.message.service.converter.QuestionConverter;
+import se.inera.intyg.certificateservice.application.message.service.validator.DeleteAnswerRequestValidator;
+import se.inera.intyg.certificateservice.domain.certificate.repository.CertificateRepository;
+import se.inera.intyg.certificateservice.domain.message.model.MessageId;
+import se.inera.intyg.certificateservice.domain.message.repository.MessageRepository;
+import se.inera.intyg.certificateservice.domain.message.service.DeleteAnswerDomainService;
+
+@Service
+@RequiredArgsConstructor
+public class DeleteAnswerService {
+
+  private final DeleteAnswerRequestValidator deleteAnswerRequestValidator;
+  private final ActionEvaluationFactory actionEvaluationFactory;
+  private final DeleteAnswerDomainService deleteAnswerDomainService;
+  private final QuestionConverter questionConverter;
+  private final CertificateRepository certificateRepository;
+  private final MessageRepository messageRepository;
+
+  @Transactional
+  public DeleteAnswerResponse delete(DeleteAnswerRequest request, String messageId) {
+    deleteAnswerRequestValidator.validate(request, messageId);
+
+    final var actionEvaluation = actionEvaluationFactory.create(
+        request.getUser(),
+        request.getUnit(),
+        request.getCareUnit(),
+        request.getCareProvider()
+    );
+
+    final var message = messageRepository.getById(
+        new MessageId(messageId)
+    );
+
+    final var certificate = certificateRepository.getById(
```

Comment:
Should we not get the certificate using the domain service so it's validated that the user is allowed to access the certificate? Or is it enough that the delete action itself is validated in its domain service?

Reply:
I think its best to let the DeleteAnswerService get the certificate from the infra layer instead of the domain layer. 

Reason for this beeing that the domain service in addition to validate if the user has access, also logs a read event, and that is not correct in this context.

I think its fine to let the DeleteAnswerDomainService determine if the delete action itself is validated or not, and not combine different action validators.